Subject: relax badshell tilde test
Author: Faidon Liambotis <paravoid@debian.org>
Bug: http://issues.asterisk.org/view.php?id=14718
Last-Update: 2009-12-19

badshell is disallowing tildes (~) in the DESTDIR directory to warn people
who use a shell that doesn't expand it to $HOMEDIR.

However, we have such a tilde in our directory (~dfsg) that isn't expanded
to $HOMEDIR nor shouldn't be.

This patch relaxes this test just for cases where the ~ is followed by a '/',
which should catch most cases.

Index: asterisk-1.8.7.2/Makefile
===================================================================
--- asterisk-1.8.7.2.orig/Makefile	2011-12-14 08:03:50.433262965 -0500
+++ asterisk-1.8.7.2/Makefile	2011-12-14 08:03:55.163265676 -0500
@@ -598,7 +598,7 @@
 	fi
 
 badshell:
-ifneq ($(findstring ~,$(DESTDIR)),)
+ifneq ($(findstring ~/,$(DESTDIR)),)
 	@echo "Your shell doesn't do ~ expansion when expected (specifically, when doing \"make install DESTDIR=~/path\")."
 	@echo "Try replacing ~ with \$$HOME, as in \"make install DESTDIR=\$$HOME/path\"."
 	@exit 1
